filebeat/filebeat.full.yml

```diff
@@ -395,22 +395,206 @@ output.elasticsearch:
   # Configure curve types for ECDHE based cipher suites
   #tls.curve_types: []
 
+#------------------------------- Kafka output ---------------------------------
+#output.kafka:
+  # The list of Kafka broker addresses to connect to.
+  #hosts: ["localhost:9092"]
+
+  # The Kafka topic used for produced events. If use_type is set to true, the
+  # topic will not be used.
+  #topic: beats
+
+  # Set Kafka topic by event type. If use_type is false, the topic option must
+  # be configured. The default is false.
+  #use_type: false
+
+  # The number of concurrent load-balanced Kafka output workers.
+  #worker: 1
+
+  # The number of times to retry publishing an event after a publishing failure.
+  # After the specified number of retries, the events are typically dropped.
+  # Some Beats, such as Filebeat, ignore the max_retries setting and retry until
+  # all events are published.  Set max_retries to a value less than 0 to retry
+  # until all events are published. The default is 3.
+  #max_retries: 3
+
+  # The maximum number of events to bulk in a single Kafka request. The default
+  # is 2048.
+  #bulk_max_size: 2048
+
+  # The number of seconds to wait for responses from the Kafka brokers before
+  # timing out. The default is 30s.
+  #timeout: 30s
+
+  # The maximum duration a broker will wait for number of required ACKs. The
+  # default is 10s.
+  #broker_timeout: 10s
+
+  # Per Kafka broker number of messages buffered in output pipeline. The default
+  # is 256.
+  #channel_buffer_size: 256
+
+  # The keep-alive period for an active network connection. If 0s, keep-alives
+  # are disabled. The default is 0 seconds.
+  #keep_alive: 0
+
+  # Sets the output compression codec. Must be one of none, snappy and gzip. The
+  # default is snappy.
+  #compression: snappy
```

Comment:
default is gzip. You have to jump some (small) hoops to get snappy support in your kafka brokers.